scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
```

Comment:
Using warnings is OK, although they should use a suitable category, for example `category=scipy.optimize.OptimizeWarning` so that the user can silence them.